test/unit/uk/gov/hmrc/apinotificationpull/controllers/ServiceLocatorConnectorSpec.scala

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.apinotificationpull.controllers
+
+
+import org.mockito.ArgumentMatchers.{any, eq => eqs}
+import org.mockito.Mockito._
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.mockito.MockitoSugar
+import play.api.libs.json.Writes
+import play.api.{Configuration, Environment}
+import uk.gov.hmrc.apinotificationpull.config.ServiceConfiguration
+import uk.gov.hmrc.apinotificationpull.connectors.{Registration, ServiceLocatorConnector}
+import uk.gov.hmrc.http.{HeaderCarrier, HttpReads, HttpResponse}
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+import uk.gov.hmrc.play.test.UnitSpec
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class ServiceLocatorConnectorSpec  extends UnitSpec with MockitoSugar with ScalaFutures {
+
+  trait Setup {
+    implicit val hc: HeaderCarrier = HeaderCarrier()
+    val serviceLocatorException = new RuntimeException
+
+    val mockHttpClient = mock[HttpClient]
+    val mockConfig = mock[ServiceConfiguration]
+    val appUrl = "http://api-notification-pull.service"
+    val appName = "api-notification-pull"
+    val serviceUrl = "https://SERVICE_LOCATOR"
+    class TestConfig extends ServiceConfiguration(mock[Configuration],mock[Environment]){
+
+      override def getString(key: String): String = key match {
+        case "appUrl" => appUrl
+        case "appName" => appName
+      }
+      override def baseUrl(serviceName: String): String = {
+        serviceUrl
+      }
+    }
+    val connector: ServiceLocatorConnector = new ServiceLocatorConnector(mockHttpClient, new TestConfig) {
+      override val handlerOK: () => Unit = mock[() => Unit]
+      override val handlerError: Throwable => Unit = mock[(Throwable) => Unit]
+      override val metadata: Option[Map[String, String]] = Some(
+        Map("third-party-api" -> "true"))
+    }
+  }
+
+  "register" should {
+    "register the JSON API Definition into the Service Locator" in new Setup {
+
+      val registration =
+        Registration(serviceName = "api-notification-pull",
+          serviceUrl = "http://api-notification-pull.service",
+          metadata = Some(Map("third-party-api" -> "true")))
+
+      when(mockHttpClient.POST(
+        eqs(s"$serviceUrl/registration"),
+        eqs(registration),
+        eqs(Seq("Content-Type" -> "application/json"))
```

Comment:
You could use the Play constants here for the content type header name and value